src/Form/formatters.js

```diff
@@ -104,10 +104,125 @@ const getCountryName = (el, options = {}) => {
     return localisedCountryName
 }
 
+/**
+ * Try to get a map of localised country names to code, or falls back to the English map
+ * @param {HTMLInputElement | HTMLSelectElement} el
+ */
+const getLocalisedCountryNamesToCodes = (el) => {
+    if (typeof Intl.DisplayNames !== 'function') return COUNTRY_NAMES_TO_CODES
+
+    // Try to infer the field language or fallback to en
+    const elLocale = inferElementLocale(el)
+
+    return Object.fromEntries(
+        Object.entries(COUNTRY_CODES_TO_NAMES)
+            .map(([code]) => [getCountryDisplayName(elLocale, code), code])
+    )
+}
+
+/**
+ * Try to infer a country code from an element we identified as identities.addressCountryCode
+ * @param {HTMLInputElement | HTMLSelectElement} el
+ * @return {string}
+ */
+const inferCountryCodeFromElement = (el) => {
+    if (COUNTRY_CODES_TO_NAMES[el.value]) return el.value
+    if (COUNTRY_NAMES_TO_CODES[el.value]) return COUNTRY_NAMES_TO_CODES[el.value]
+
+    const localisedCountryNamesToCodes = getLocalisedCountryNamesToCodes(el)
+    if (localisedCountryNamesToCodes[el.value]) return localisedCountryNamesToCodes[el.value]
+
+    if (el instanceof HTMLSelectElement) {
+        const selectedText = el.selectedOptions[0].text
+        if (COUNTRY_CODES_TO_NAMES[selectedText]) return selectedText
+        if (COUNTRY_NAMES_TO_CODES[selectedText]) return localisedCountryNamesToCodes[selectedText]
+        if (localisedCountryNamesToCodes[selectedText]) return localisedCountryNamesToCodes[selectedText]
+    }
+    return ''
+}
+
+/** @param {InternalDataStorageObject} credentials */
+const shouldStoreCredentials = ({credentials}) =>
+    credentials.password
+
+/** @param {InternalDataStorageObject} credentials */
+const shouldStoreIdentities = ({identities}) =>
+    (identities.firstName || identities.fullName) && identities.addressStreet && identities.addressCity
+
+/** @param {InternalDataStorageObject} credentials */
+const shouldStoreCreditCards = ({creditCards}) =>
+    creditCards.cardNumber && creditCards.cardSecurityCode
+
+/**
+ * Formats form data into an object to send to the device for storage
+ * If values are insufficient for a complete entry, they are discarded
+ * @param {InternalDataStorageObject} formValues
+ * @return {DataStorageObject}
+ */
+const prepareFormValuesForStorage = (formValues) => {
+    let {credentials, identities, creditCards} = formValues
+
+    /** Fixes for credentials **/
+    // Don't store if there isn't enough data
+    if (shouldStoreCredentials(formValues)) {
+        // If we don't have a username to match a password, let's see if the email is available
+        if (credentials.password && !credentials.username && identities.emailAddress) {
+            credentials.username = identities.emailAddress
+        }
+    } else {
+        // @ts-ignore
```

Comment:
Note: It may be that instead of `Partial`, you might want to change the raw definition of the type in question, to have optional fields instead? I'll let you make the decision on which :)